sql/insert.go

```diff
@@ -186,3 +224,94 @@ func (p *planner) processColumns(tableDesc *TableDescriptor,
 
 	return cols, nil
 }
+
+func (p *planner) makeDefaultExprs(cols []ColumnDescriptor) ([]parser.Expr, error) {
+	// Check to see if any of the columns have DEFAULT expressions. If there are
+	// no DEFAULT expressions, we don't bother with constructing the defaults map
+	// as the defaults are all NULL.
+	haveDefaults := false
+	for _, col := range cols {
+		if col.DefaultExpr != nil {
+			haveDefaults = true
+			break
+		}
+	}
+	if !haveDefaults {
+		return nil, nil
+	}
+
+	// Construct a SELECT statement with an output for each column. The output is
+	// either the DEFAULT expression or NULL if no default expression was
+	// configured.
+	var buf bytes.Buffer
+	buf.WriteString("SELECT ")
+	for i, col := range cols {
+		if i > 0 {
+			buf.WriteString(", ")
+		}
+		if col.DefaultExpr != nil {
+			buf.WriteString(*col.DefaultExpr)
+		} else {
+			buf.WriteString("NULL")
+		}
+	}
+
+	sql := buf.String()
+	stmts, err := parser.Parse(sql, parser.Syntax(p.session.Syntax))
```

Comment:
We check `parser.ContainsVars` when executing the DDL statements, but we don't repeat this validation when evaluating the defaults. If an expression parses differently between the different syntax modes then the use of `p.session.Syntax` could have allowed a variable reference to sneak through. It would probably good to repeat that check in any case to guard against any potential bugs in our parse-stringify-parse cycle.

We also mentioned subqueries as something that shouldn't be allowed in defaults, but I don't see any enforcement or testing of that in this change.

SQL injection admittedly isn't a major concern here - any attacker who can get something tricky installed as a default expression has probably won the game and doesn't need any further holes.